### DISCUSSION TITLE
Can I use one single element, instead of different control, for everything I need to do

### DISCUSSION THREAD

**Discussion by Xyncgas** [AvaloniaUI/Avalonia#10250] [General]
For example, web has the 'div' element, you can attach eventlisteners to it and it tells you when it's clicked for example, you can do different things from then on.

I appreciate Avalonia having so many controls, for example buttons, I personally am very comfortable with having one single empty thing, that I can use absolute position to put it where I want, style it how I want, it would be nicer if I can stick with this one element instead of going for a button or another control for things I need to do I want to attach events / call backs to what happens.

This way it creates much less noice, I just describe the element, and Avalonia put the component on screen which it can be clicked for example to generate the same thing with a different layout. Having this one single 'unified' element makes prototyping so fast because I can template it myself to show a red box for all the elements I am putting on screen, I can go head and implement menu and tool box and all the things my application wants to do with the fluidity and the lack of noise and when you are using Avalonia with F# the code is going to be looking so clean because I don't have to style it or put text in the element

**Comment by maxkatz6** [comment]
You can use something like Border for everything and handle all the input and styling manually. But I doubt it will be clean or good. 

Neither "div for for everything"  is a good approach in web.  That's exactly why they are using components (web components or react/vue/angular components), hiding plain divs inside.
